### PR TITLE
Add NumPy 2.x CI for Windows

### PR DESCRIPTION
**Pull request by asi1024**
Part of https://github.com/cupy/cupy/issues/8306 and https://github.com/cupy/cupy/issues/8514.

### CODE REVIEW

File: cupy/_functional/vectorize.py [hunk 1]
```diff
@@ -6,13 +6,10 @@
...
-        return 'l'
-    if isinstance(arg, float):
-        return 'd'
-    if isinstance(arg, complex):
-        return 'D'
-    return arg.dtype.char
+    if hasattr(arg, 'dtype'):
+        return arg.dtype.char
+    else:
+        return numpy.dtype(type(arg)).char
```
Comment (kmaehashi) [inline cupy/_functional/vectorize.py:12]: This line seems like a slower replacement of the existing one?

File: cupy/random/_generator.py [hunk 2]
```diff
@@ -212,12 +212,14 @@ def geometric(self, p, size=None, dtype=int):
         p = cupy.asarray(p)
         if size is None:
             size = p.shape
+        if dtype is None:
+            dtype = numpy.dtype('l')
```
Comment (kmaehashi) [inline cupy/random/_generator.py:216]: Can we avoid creating a new dtype instance everytime? (or use `numpy.int_`?)

File: cupy/_functional/vectorize.py (hunk 1 above)
Comment (asi1024) [reply cupy/_functional/vectorize.py:12]: That returns `l` on Linux, `q` on Widnows.

File: cupy/random/_generator.py (hunk 2 above)
Comment (asi1024) [reply cupy/random/_generator.py]: Replaced with `numpy.long`. (`numpy.int_` is `q` on Windows)

**Review by kmaehashi — APPROVED** [review]

**Comment by asi1024** [comment]
cupy.win.cuda112 timeout seems unrelated.

**Pull request merged**